--- origsrc/src/Makefile	2008-05-23 07:26:29.000000000 -0500
+++ src/src/Makefile	2008-05-23 07:36:02.281250000 -0500
@@ -7,6 +7,8 @@
 all: libnut nututils
 
 libnut: libnut/libnut.a
+libnut-shared: libnut/libnut.so
+libnut-dll: libnut/cygnut-0.dll
 
 libnut/libnut.a: $(LIBNUT_OBJS)
 	rm -f $@
@@ -16,6 +18,9 @@
 libnut/libnut.so: $(LIBNUT_OBJS)
 	$(CC) $(CFLAGS) -shared $^ -o $@
 
+libnut/cygnut-0.dll: $(LIBNUT_OBJS)
+	${CC} ${CFLAGS} ${LDFLAGS} -shared -Wl,--enable-auto-image-base,--out-implib,libnut/libnut.dll.a $^ -o $@
+
 $(LIBNUT_OBJS): libnut/priv.h libnut/libnut.h
 
 nututils: $(NUTUTILS_PROGS)
@@ -35,6 +40,12 @@
 	install -d $(prefix)/lib
 	install -m 644 libnut/libnut.so $(prefix)/lib
 
+install-libnut-dll: libnut/cygnut-0.dll install-libnut-headers
+	install -d $(prefix)/bin
+	install -d $(prefix)/lib
+	install -m 644 libnut/cygnut-0.dll $(prefix)/bin
+	install -m 644 libnut/libnut.dll.a $(prefix)/lib
+
 install-libnut-headers:
 	install -d $(prefix)/include
 	install -m 644 libnut/libnut.h $(prefix)/include
